src/vcpkg/base/messages.cpp

```diff
@@ -0,0 +1,288 @@
+#include <vcpkg/base/json.h>
+#include <vcpkg/base/messages.h>
+#include <vcpkg/base/system.debug.h>
+
+namespace vcpkg::msg
+{
+    DECLARE_AND_REGISTER_MESSAGE(NoLocalizationForMessages, (), "", "No localization for the following messages:");
+
+    // basic implementation - the write_unlocalized_text_to_stdout
+#if defined(_WIN32)
+    static bool is_console(HANDLE h)
+    {
+        DWORD mode = 0;
+        // GetConsoleMode succeeds iff `h` is a console
+        // we do not actually care about the mode of the console
+        return GetConsoleMode(h, &mode);
+    }
+
+    static void check_write(BOOL success)
+    {
+        if (!success)
+        {
+            ::fwprintf(stderr, L"[DEBUG] Failed to write to stdout: %lu\n", GetLastError());
+            std::abort();
+        }
+    }
+    static DWORD size_to_write(::size_t size)
+    {
+        return size > static_cast<DWORD>(-1) // DWORD_MAX
+                   ? static_cast<DWORD>(-1)
+                   : static_cast<DWORD>(size);
+    }
+
+    void write_unlocalized_text_to_stdout(Color c, StringView sv)
+    {
+        if (sv.empty()) return;
+
+        static const HANDLE stdout_handle = ::GetStdHandle(STD_OUTPUT_HANDLE);
+        static const bool stdout_is_console = is_console(stdout_handle);
+
+        if (stdout_is_console)
+        {
+            WORD original_color = 0;
+            if (c != Color::none)
+            {
+                CONSOLE_SCREEN_BUFFER_INFO console_screen_buffer_info{};
+                ::GetConsoleScreenBufferInfo(stdout_handle, &console_screen_buffer_info);
```

Comment:
Sorry if I missed this before but I believe there is missing error handling here.

Reply:
Should we not change the color if the console color is strange? (for example, if the user's console background color is yellow, this might have us writing yellow on a yellow background)

I know this is probably preexisteing but still wanted to ask.

Reply:
on the latter point - I don't want to change existing behavior here, at least for now. We haven't gotten complaints yet.